### ISSUE TITLE
no-unstable-nested-components does not detect components starting with underscore

### ISSUE THREAD

**Issue by WickyNilliams** [yannickcr/eslint-plugin-react#4002]
A pattern we use commonly in our codebase (pre-dates me, and we're too far along the road now to change) is to prefix some components with an underscore e.g. when forwarding refs, or to signal a component shouldn't be exported etc.

Components with this naming convention do not get detected as parent components, and thus the unstable nested components are not flagged

```tsx
function _Parent() {
  function NotFlagged() {
    return <b>oh no</b>
  }

  return <NotFlagged />
}
```

Given there have been other issues raised relating, and carve outs added for, this naming convention:

* https://github.com/jsx-eslint/eslint-plugin-react/issues/3560
* https://github.com/jsx-eslint/eslint-plugin-react/issues/2603

I think this constitutes a bug/oversight in this rule. Happy to make a PR, if there is agreement

**Comment by ljharb** [comment]
While I think a codemod would be trivial to change this pattern, I agree that since React supports a leading underscore, so too must we.
